server/database.go

```diff
@@ -501,14 +501,14 @@ func (s *Server) registerDatabaseRoutes(g *echo.Group) {
 		}
 
 		if err := createBackupDirectory(s.profile.DataDir, database.ID); err != nil {
-			return err
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to create directory in %q to store backup of database %q", s.profile.DataDir, database.Name)).SetInternal(err)
 		}
 		path := getBackupRelativeFilePath(database.ID, backupCreate.Name)
 		backupCreate.Path = path
 
 		driver, err := getAdminDatabaseDriver(ctx, database.Instance, database.Name, s.pgInstanceDir)
 		if err != nil {
-			return err
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to connect to instance %q", database.Instance.Name)).SetInternal(err)
```

Comment:
this should not be an internal error since it's most likely a misconfig connection.

Reply:
How about [400 Bad Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400)?

> The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).

IMO, it's due to a bad config of the instance from the client (user).

Reply:
How about [504 Gateway Timeout](https://datatracker.ietf.org/doc/html/rfc7231#section-6.6.5)?
>  The 504 (Gateway Timeout) status code indicates that the server,
   while acting as a gateway or proxy, did not receive a timely response
   from an upstream server it needed to access in order to complete the
   request.

Reply:
I don't think 504 is appropriate.
504 Gateway Timeout is used if you are nginx and cannot receive a response from the upstream application server. So you tell the client that you can do nothing because nobody is responding to you in time.

Reply:
Also prefer 400, there is no perfect error code, 400 is a less evil one.

Reply:
When errors are unclear, I also prefer to use client errors with 4xx.